autoscheduler/frontend/src/tests/Redux.test.ts

```diff
@@ -183,3 +188,152 @@ test('Updates course card boolean field', () => {
   // assert
   expect(store.getState().courseCards[0].web).toBeTruthy();
 });
+
+test('Merges overlapping availabilities of same type', () => {
+  // arrange
+  const store = createStore(autoSchedulerReducer);
+
+  // act
+  store.dispatch(addAvailability({
+    available: AvailabilityType.BUSY,
+    dayOfWeek: 2,
+    time1: 8 * 60 + 0,
+    time2: 12 * 60 + 0,
+  }));
+  store.dispatch(addAvailability({
+    available: AvailabilityType.BUSY,
+    dayOfWeek: 2,
+    time1: 11 * 60 + 30,
+    time2: 13 * 60 + 42,
+  }));
+
+  // assert
+  expect(store.getState().availability).toEqual([{
+    available: AvailabilityType.BUSY,
+    dayOfWeek: 2,
+    startTimeHours: 8,
+    startTimeMinutes: 0,
+    endTimeHours: 13,
+    endTimeMinutes: 42,
+  }]);
+});
+
+test('Doesn\'t merge non-overlapping availabilities', () => {
+  // arrange
+  const store = createStore(autoSchedulerReducer);
+  const availability1 = {
+    available: AvailabilityType.BUSY,
+    dayOfWeek: 2,
+    time1: 8 * 60 + 0,
+    time2: 12 * 60 + 0,
+  };
+  const availability2 = {
+    available: AvailabilityType.BUSY,
+    dayOfWeek: 2,
+    time1: 12 * 60 + 30,
+    time2: 13 * 60 + 42,
+  };
+
+  // act
+  store.dispatch(addAvailability(availability1));
+  store.dispatch(addAvailability(availability2));
+
+  // assert
+  expect(store.getState().availability).toEqual(
+    [argsToAvailability(availability1), argsToAvailability(availability2)],
+  );
+});
+
+test('Merging availabilities works with overlaps on both ends', () => {
+  // arrange
+  const store = createStore(autoSchedulerReducer);
+
+  // act
+  store.dispatch(addAvailability({
+    available: AvailabilityType.BUSY,
+    dayOfWeek: 2,
+    time1: 11 * 60 + 30,
+    time2: 13 * 60 + 42,
+  }));
+  store.dispatch(addAvailability({
+    available: AvailabilityType.BUSY,
+    dayOfWeek: 2,
+    time1: 7 * 60 + 0,
+    time2: 8 * 60 + 30,
+  }));
+  store.dispatch(addAvailability({
+    available: AvailabilityType.BUSY,
+    dayOfWeek: 2,
+    time1: 8 * 60 + 0,
+    time2: 12 * 60 + 0,
+  }));
+
+  // assert
+  expect(store.getState().availability).toEqual([{
+    available: AvailabilityType.BUSY,
+    dayOfWeek: 2,
+    startTimeHours: 7,
+    startTimeMinutes: 0,
+    endTimeHours: 13,
+    endTimeMinutes: 42,
+  }]);
+});
+
+test('Deletes availbility', () => {
+  // arrange
+  const store = createStore(autoSchedulerReducer);
+
+  // act
+  store.dispatch(addAvailability({
+    available: AvailabilityType.BUSY,
+    dayOfWeek: 2,
+    time1: 8 * 60 + 0,
+    time2: 12 * 60 + 0,
+  }));
+  const intermediateState = store.getState().availability;
+  store.dispatch(deleteAvailability({
+    available: AvailabilityType.BUSY,
+    dayOfWeek: 2,
+    time1: 8 * 60 + 0,
+    time2: 12 * 60 + 0,
```

Comment:
Since you're using the same data twice, I'd say create the availability dict in arrange. Also think this will help with clarity of the test in general, since you'll see that you're adding and deleting the same object w/o having to check all of the individual fields.

Reply:
Honestly I think this same sentiment applies to a bunch of the tests. For tests where the dicts aren't the same, extract out the constants that you're setting in every availability object (i.e. `dayOfWeek` and `available`).

Also, do it for times in the cases that they're small differences of each other (since the specific times in most cases don't really matter, it's their relation to the other times that do, i.e. for testing merging).